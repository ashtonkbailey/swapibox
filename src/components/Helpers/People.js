const fetchPropertyObj = async (url) => {
  const response = await fetch(url);
  const propertyObj = await response.json();
  return propertyObj;
};

const createDataSet = async data => Promise.all(data.results.map(async (person, index) => {
  const currHomeworld = await fetchPropertyObj(person.homeworld);
  const currSpecies = await fetchPropertyObj(person.species[0]);
  return {
    name: person.name,
    homeworld: currHomeworld.name,
    population: currHomeworld.population,
    species: currSpecies.name,
    index,
    favorite: false,
  };
}));

const People = async () => {
  const url = 'https://swapi.co/api/people/';
  const response = await fetch(url);
  const data = await response.json();
  return createDataSet(data);
};

export default People;
