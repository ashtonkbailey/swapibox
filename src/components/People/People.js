import React, { Component } from 'react';
import '../PeopleCard/peopleCard.css';
import lightsabers from '../../images/lightsabers.png';
import luke from '../../images/luke.jpg'

const People = ({carouselIndex, addToFavorites}) => {
  let hiddenClass;
  const leftCardIndex = carouselIndex;
  let middleCardIndex = carouselIndex + 1;
  let rightCardIndex = carouselIndex + 2;

  if (carouselIndex === 9) {
    middleCardIndex = 0;
    rightCardIndex = carouselIndex - 8;
  }

  if (carouselIndex === 8) {
    rightCardIndex = carouselIndex - 8;
  }
  
  const peopleData = JSON.parse(localStorage.getItem('people'));
  const peopleDisplay = peopleData.map(person => {
    switch (person.index) {
      case middleCardIndex:
        hiddenClass = 'middle-card';
        break;
      case leftCardIndex:
        hiddenClass = 'left-card small-card';
        break;
      case rightCardIndex:
        hiddenClass = 'right-card small-card';
        break;
      default:
        hiddenClass = 'hidden';
    }
    return (<article className={hiddenClass} key={person.name}>
        <h3 className="name">
          {person.name}
        </h3>
        <img
          src={luke}
          alt="character"
        />
        <p>
          species:
          {person.species}
        </p>
        <p>
          homeworld:
          {person.homeworld}
        </p>
        <p>
          population:
          {person.population}
        </p>
        <img
          className="saber"
          src={lightsabers}
          onClick={() => addToFavorites( person )}
          alt="click to add to favorites"
        />
      </article>)
  });
  
  return (
     <div>
       {peopleDisplay}
     </div>
    
  )
}


export default People;