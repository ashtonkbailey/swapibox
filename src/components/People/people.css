article {
  align-items: center;
  color:  #FBF230;
  display: flex;
  flex-direction: column;
  font-size: 16px;
  padding: 6px;
  transition: background-color, opacity .6s ease-in-out;
}

.picture {
  border-radius: 20px;
  border: solid 1px white;
  margin: 10px;
  margin: 1em;
  max-height: 200px;
}

.middle-card>.picture {
  width: 40%;
  height: 200px;
}

.middle-card>.picture {
  width: 40%;
  height: 200px;
}

.saber {
  bottom: 16px;
  filter: grayscale(40%);
  height: 60px;
  margin-top: 10px;
  outline: none;
  position: relative;
  width: 18%;
}

.saber:hover {
  cursor: pointer;
  filter: constrast(200%);
  transform: scale(1.1);
  transition: .3s ease-in-out;
}

.saber.true {
  filter: contrast(200%);
}

.name {
  color: black;
  font-family: Roboto;
  font-size: 40px;
  font-style: normal;
  margin: 0;
  text-align: center;
  text-shadow:
  -1px -1px 0 rgb(238, 213, 19),  
  1px -1px 0 rgb(238, 213, 19),
  -1px 1px 0 rgb(238, 213, 19),
  1px 1px 0 rgb(238, 213, 19);
}

.small-card .name {
  font-size:  20px; 
}

.small-card {
  background: rgba(227, 218, 135, 0.2);
  height: 500px;
  margin-top: 60px;
  pointer-events: none;
  transition: background-color, opacity .6s ease-in-out;
}

.small-card>.saber {
  width: 60px;
}

.left-card {
  left: 5%;
  opacity: .5;
  position: fixed;
  width: 22%;
}

.right-card {
  opacity: .5;
  position: fixed;
  right: 5%;
  width: 22%;
}


.middle-card {
  background: rgba(227, 218, 135, 0.4);
  border-radius: 40px;
  border: 4px solid #EDE748;
  box-shadow: 0px 4px 40px rgb(227, 218, 135);
  font-size: 24px;
  height: 60%;
  left: 32%;
  margin: 20px;
  position: fixed;
  width: 34%;
}

.small-card>img {
  width: 32%;
}

.middle-card>.saber {
  height: 16%;
  width: 16%;
}

.luke {
  border-radius: 20px;
  border: solid 1px black;
  box-shadow: 2px 2px  black;
  margin: 2px;
}

.left-card, .right-card {
  background: rgba(0, 0, 0, 0.66);
  border-radius: 40px;
  border: 2px solid #EDE748;
  height: 400px;
}

p {
  color: black;
  margin: 10px;
  text-shadow: 
  .5px .5px 0 rgb(238, 213, 19);
}