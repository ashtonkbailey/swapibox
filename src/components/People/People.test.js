import People from './People.js';
import React from 'react';
import { mount, shallow } from 'enzyme';

describe ('people',() => {
  let mockCard;
  let mockIndex;
  let wrapper;
  let mockAddToFavorites;
  let peopleData;
 

  beforeEach(() => {
    mockCard = {
      favorite: false,
      homeworld: 'Tatooine',
      index: 0,
      name: 'Luke Skywalker',
      population: '200000',
      species: 'Human',
    };
    mockAddToFavorites = jest.fn();
    wrapper = shallow (<People carouselIndex={mockIndex} addToFavorites={mockAddToFavorites}/>);
    peopleData = [
      mockCard,
      {
        favorite: false,
        homeworld: 'Tatooine',
        index: 1,
        name: 'C-3PO',
        population: '200000',
        species: 'Droid',
      },
      {
        favorite: false,
        homeworld: 'Naboo',
        index: 2,
        name: 'R2-D2',
        population: '4500000000',
        species: 'Droid',
      },
    ];
  })
  it('matches the snpashot with all parameters passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })
})