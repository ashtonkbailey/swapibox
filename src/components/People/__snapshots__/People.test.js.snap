// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people matches the snpashot with all parameters passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    addToFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="hidden"
        >
          <h3
            className="name"
          >
            Luke Skywalker
          </h3>
          <img
            alt="character"
            className="picture"
            src="./luke.jpg"
          />
          <p>
            species:
            Human
          </p>
          <p>
            homeworld:
            Tatooine
          </p>
          <p>
            population:
            200000
          </p>
          <img
            alt="click to add to favorites"
            className="saber false"
            onClick={[Function]}
            src="lightsabers.png"
          />
        </article>,
        <article
          className="hidden"
        >
          <h3
            className="name"
          >
            R2-D2
          </h3>
          <img
            alt="character"
            className="picture"
            src="./R2D2.jpeg"
          />
          <p>
            species:
            Droid
          </p>
          <p>
            homeworld:
            Naboo
          </p>
          <p>
            population:
            4500000000
          </p>
          <img
            alt="click to add to favorites"
            className="saber false"
            onClick={[Function]}
            src="lightsabers.png"
          />
        </article>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="name"
            >
              Luke Skywalker
            </h3>,
            <img
              alt="character"
              className="picture"
              src="./luke.jpg"
            />,
            <p>
              species:
              Human
            </p>,
            <p>
              homeworld:
              Tatooine
            </p>,
            <p>
              population:
              200000
            </p>,
            <img
              alt="click to add to favorites"
              className="saber false"
              onClick={[Function]}
              src="lightsabers.png"
            />,
          ],
          "className": "hidden",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
              "className": "name",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "character",
              "className": "picture",
              "src": "./luke.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "species:",
                "Human",
              ],
            },
            "ref": null,
            "rendered": Array [
              "species:",
              "Human",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "homeworld:",
                "Tatooine",
              ],
            },
            "ref": null,
            "rendered": Array [
              "homeworld:",
              "Tatooine",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population:",
                "200000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "population:",
              "200000",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "click to add to favorites",
              "className": "saber false",
              "onClick": [Function],
              "src": "lightsabers.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "R2-D2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="name"
            >
              R2-D2
            </h3>,
            <img
              alt="character"
              className="picture"
              src="./R2D2.jpeg"
            />,
            <p>
              species:
              Droid
            </p>,
            <p>
              homeworld:
              Naboo
            </p>,
            <p>
              population:
              4500000000
            </p>,
            <img
              alt="click to add to favorites"
              className="saber false"
              onClick={[Function]}
              src="lightsabers.png"
            />,
          ],
          "className": "hidden",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "R2-D2",
              "className": "name",
            },
            "ref": null,
            "rendered": "R2-D2",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "character",
              "className": "picture",
              "src": "./R2D2.jpeg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "species:",
                "Droid",
              ],
            },
            "ref": null,
            "rendered": Array [
              "species:",
              "Droid",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "homeworld:",
                "Naboo",
              ],
            },
            "ref": null,
            "rendered": Array [
              "homeworld:",
              "Naboo",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population:",
                "4500000000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "population:",
              "4500000000",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "click to add to favorites",
              "className": "saber false",
              "onClick": [Function],
              "src": "lightsabers.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="hidden"
          >
            <h3
              className="name"
            >
              Luke Skywalker
            </h3>
            <img
              alt="character"
              className="picture"
              src="./luke.jpg"
            />
            <p>
              species:
              Human
            </p>
            <p>
              homeworld:
              Tatooine
            </p>
            <p>
              population:
              200000
            </p>
            <img
              alt="click to add to favorites"
              className="saber false"
              onClick={[Function]}
              src="lightsabers.png"
            />
          </article>,
          <article
            className="hidden"
          >
            <h3
              className="name"
            >
              R2-D2
            </h3>
            <img
              alt="character"
              className="picture"
              src="./R2D2.jpeg"
            />
            <p>
              species:
              Droid
            </p>
            <p>
              homeworld:
              Naboo
            </p>
            <p>
              population:
              4500000000
            </p>
            <img
              alt="click to add to favorites"
              className="saber false"
              onClick={[Function]}
              src="lightsabers.png"
            />
          </article>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="name"
              >
                Luke Skywalker
              </h3>,
              <img
                alt="character"
                className="picture"
                src="./luke.jpg"
              />,
              <p>
                species:
                Human
              </p>,
              <p>
                homeworld:
                Tatooine
              </p>,
              <p>
                population:
                200000
              </p>,
              <img
                alt="click to add to favorites"
                className="saber false"
                onClick={[Function]}
                src="lightsabers.png"
              />,
            ],
            "className": "hidden",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker",
                "className": "name",
              },
              "ref": null,
              "rendered": "Luke Skywalker",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "character",
                "className": "picture",
                "src": "./luke.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "species:",
                  "Human",
                ],
              },
              "ref": null,
              "rendered": Array [
                "species:",
                "Human",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "homeworld:",
                  "Tatooine",
                ],
              },
              "ref": null,
              "rendered": Array [
                "homeworld:",
                "Tatooine",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population:",
                  "200000",
                ],
              },
              "ref": null,
              "rendered": Array [
                "population:",
                "200000",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "click to add to favorites",
                "className": "saber false",
                "onClick": [Function],
                "src": "lightsabers.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "R2-D2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="name"
              >
                R2-D2
              </h3>,
              <img
                alt="character"
                className="picture"
                src="./R2D2.jpeg"
              />,
              <p>
                species:
                Droid
              </p>,
              <p>
                homeworld:
                Naboo
              </p>,
              <p>
                population:
                4500000000
              </p>,
              <img
                alt="click to add to favorites"
                className="saber false"
                onClick={[Function]}
                src="lightsabers.png"
              />,
            ],
            "className": "hidden",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "R2-D2",
                "className": "name",
              },
              "ref": null,
              "rendered": "R2-D2",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "character",
                "className": "picture",
                "src": "./R2D2.jpeg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "species:",
                  "Droid",
                ],
              },
              "ref": null,
              "rendered": Array [
                "species:",
                "Droid",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "homeworld:",
                  "Naboo",
                ],
              },
              "ref": null,
              "rendered": Array [
                "homeworld:",
                "Naboo",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population:",
                  "4500000000",
                ],
              },
              "ref": null,
              "rendered": Array [
                "population:",
                "4500000000",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "click to add to favorites",
                "className": "saber false",
                "onClick": [Function],
                "src": "lightsabers.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
